tsl/src/bgw_policy/continuous_aggregate_api.c

```diff
@@ -619,6 +727,33 @@ policy_refresh_cagg_add_internal(Oid cagg_oid, Oid start_offset_type, NullableDa
 	JsonbValue *result = pushJsonbValue(&parse_state, WJB_END_OBJECT, NULL);
 	Jsonb *config = JsonbValueToJsonb(result);
 
+	CaggOverlapResult res;
+	res = policy_refresh_cagg_check_for_overlaps(cagg, config, 0);
+	switch (res)
+	{
+		case OVERLAP_NONE:
+			break;
+		case OVERLAP_EXACT:
+			if (if_not_exists)
+			{
+				ereport(NOTICE,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("continuous aggregate policy already exists for \"%s\", "
+								"skipping",
+								get_rel_name(cagg->relid))));
+				PG_RETURN_INT32(-1);
+			}
+			/* else fallthrough and return error */
+			TS_FALLTHROUGH;
```

Comment:
```suggestion
			ereport(ERROR,
					(errcode(ERRCODE_DUPLICATE_OBJECT),
					 errmsg("continuous aggregate policy already exists for \"%s\"",
							get_rel_name(cagg_oid)),
					 errdetail("Only one continuous aggregate policy can be created per continuous "
							   "aggregate and a policy with job id %d already exists for \"%s\".",
							   ((BgwJob *) linitial(jobs))->fd.id,
							   get_rel_name(cagg_oid))));
```
I think here we should error out as before